spec/index.bs

```diff
@@ -2428,114 +2428,115 @@ A {{GPUBindGroupLayout}} object has the following internal slots:
-            , ensure [=sampler validation=] is not violated.
-        1. Insert |bindingDescriptor| into |layout|.{{GPUBindGroupLayout/[[entryMap]]}}
-            with the key of |bindingDescriptor|/{{GPUBindGroupLayoutEntry/binding}}.
-    1. Return |layout|.
-
-    <div class=validusage dfn-for=GPUDevice.createBindGroupLayout>
-        <dfn abstract-op>Valid Usage</dfn>
-
-        If any of the following conditions are violated:
-            1. Generate a {{GPUValidationError}} in the current scope with appropriate error message.
-            1. Create a new [=invalid=] {{GPUBindGroupLayout}} and return the result.
-
-        <dfn>bind group layout device validation</dfn>: The {{GPUDevice}} must not be lost.
-
-        <dfn>binding validation</dfn>: Each |bindingDescriptor|.{{GPUBindGroupLayoutEntry/binding}} in |descriptor| must be unique.
-
-        <dfn>vertex shader binding validation</dfn>: {{GPUBindingType/storage-buffer}} and {{GPUBindingType/writeonly-storage-texture}} are not allowed.
+<dl dfn-type=method dfn-for=GPUDevice>
+    : <dfn>createBindGroupLayout(descriptor)</dfn>
+    ::
 
-        <dfn>uniform buffer validation</dfn>: There must be {{GPULimits/maxUniformBuffersPerShaderStage|GPULimits.maxUniformBuffersPerShaderStage}} or
-            fewer |bindingDescriptor|s of type {{GPUBindingType/uniform-buffer}} visible on each shader stage in |descriptor|.
+    Creates a {{GPUBindGroupLayout}}.
 
-        <dfn>dynamic uniform buffer validation</dfn>: There must be {{GPULimits/maxDynamicUniformBuffersPerPipelineLayout|GPULimits.maxDynamicUniformBuffersPerPipelineLayout}} or
-                fewer |bindingDescriptor|s of type {{GPUBindingType/uniform-buffer}} with {{GPUBindGroupLayoutEntry/hasDynamicOffset}} set to `true` in
-                |descriptor| that are visible to any shader stage.
+    <div algorithm=GPUDevice.createBindGroupLayout>
+        **Called on:** {{GPUDevice}} |this|.
 
-        <dfn>storage buffer validation</dfn>: There must be {{GPULimits/maxStorageBuffersPerShaderStage|GPULimits.maxStorageBuffersPerShaderStage}} or
-            fewer |bindingDescriptor|s of type {{GPUBindingType/storage-buffer}} visible on each shader stage in |descriptor|.
+        **Arguments:**
+        <pre class=argumentdef for="GPUDevice/createBindGroupLayout(descriptor)">
+            |descriptor|:
+        </pre>
+
+        **Returns:** {{GPUBindGroupLayout}}
+
+        1. If any of the following conditions are unsatisfied:
+            <div class=validusage>
+                - |this| is a [=valid=] {{GPUDevice}}.
+                - Each |bindingDescriptor|.{{GPUBindGroupLayoutEntry/binding}} in |descriptor| is unique.
+                - There is {{GPULimits/maxUniformBuffersPerShaderStage|GPULimits.maxUniformBuffersPerShaderStage}}
+                    or fewer |bindingDescriptor|s of type {{GPUBindingType/uniform-buffer}} visible
+                    on each shader stage in |descriptor|.
```

Comment:
Could use a reword, IMO. Something like:

```suggestion
                - For each shader stage, the number of entries in |descriptor| with
                    {{GPUBindGroupLayout/type}} {{GPUBindingType/uniform-buffer}} &le;
                    {{GPULimits/maxUniformBuffersPerShaderStage|maxUniformBuffersPerShaderStage}}.
```

Reply:
SGTM. Tried to apply similar phrasing across the board.